### PR TITLE
ISPN-6695 Create CacheSet version that can easily be mapped over

### PR DESCRIPTION
**Pull request by wburns**
https://issues.jboss.org/browse/ISPN-6695

### CODE REVIEW

File: commons/src/main/java/org/infinispan/commons/util/DistinctFunction.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * the function returns distinct values.  This can be helpful because when a function is applied to data is ambiguous
+ * if the resulting data produced is distinct or not.  This allows some callers of this method to apply additional
+ * performance optimizations taking this into account.
+ * <p>
+ * If a <b>function</b> is implemented with this and it doesn't produce distinct values, the operation of the
+ * consumer of this function may be undefined.
+ * @author wburns
+ * @since 9.0
+ */
+public interface DistinctFunction<T, R> {
```
Comment (wburns) [inline commons/src/main/java/org/infinispan/commons/util/DistinctFunction.java]: Sorry I snuck this one in cause I will want it for dist sorting and it makes sense here too :)

**Comment by galderz** [comment]
Looking... :)

**Comment by galderz** [comment]
Looks great, integrating... :)

**Comment by galderz** [comment]
Integrated, thanks @wburns 😃

**Pull request closed without merging**